lib/camp.js

```diff
@@ -106,6 +106,13 @@ function augmentReqRes(req, res, server) {
       res.end('404\n');
     });
   };
+  // Sending JSON data
+  res.json = function (data) {
+    res.setHeader('Content-Type', mime.json);
+    // This gets compressed anyway, so we might as well make it human-readable
+    var json = JSON.stringify(data, null, 2);
+    getCompressedStream(req, res).end(json);
+  };
```

Comment:
Actually `res.end({foo: 'bar'})` doesn't work, because there is no `res.end` augmentation (only [res.template and res.file](https://github.com/espadrine/sc/blob/master/lib/camp.js#L71)), so it uses Node.js's `response.end(chunk)`, and `chunk` needs to be [a string or a buffer](https://nodejs.org/api/http.html#http_request_write_chunk_encoding_callback).

I tested it, and `res.end({foo: 'bar'})` makes the connection hang.

Also, if you're actually referring to the `end` callback in `camp.route`, this will only try to template the given `data` [into the requested resource](https://github.com/espadrine/sc/blob/master/lib/camp.js#L845).

I think the desired feature doesn't exist yet. Are you ok with my `res.json` implementation, or would you rather have the solution implemented differently, e.g. in an augmented `res.end`? (FWIW express has a `res.json()`, and I think such a thing plays nicely next to the existing `res.template()` and `res.file()` helpers).